### PR TITLE
update goal statuses based on AR state per Jira

### PR DESCRIPTION
**Pull request by hardwarehuman**
## Description of change

Update the affected goals so that any of them that is not on an approved AR is set to Not Started and any that _is_ on an approved AR is set to In Progress regardless of their previous status, including if they were closed.

Results:
```
       operation        | cnt
------------------------+-----
 updated_to_in_progress |   8
 updated_to_not_started |   1
 ```

## How to test

Run migration

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2093
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2203

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/migrations/20231017230001-fei-goal-status-changes.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            (55635), (56022), (55115), (56047), (54317)
+        ) AS data(gid)
+      )
+      SELECT
+        gid,
+        BOOL_OR(ar.id IS NOT NULL) on_approved_ar
+      FROM goalset g
+      JOIN "Objectives" o
+        ON o."goalId" = g.gid
+      LEFT JOIN "ActivityReportObjectives" aro
```
Comment (AdamAdHocTeam) [inline src/migrations/20231017230001-fei-goal-status-changes.js:36]: Why don't we go through ActivityReportGoals here (instead of objectives)?

**Review by AdamAdHocTeam — COMMENTED** [review]
In my DB (from a few weeks ago) I didn't see these goals linked to any AR's with the calculatedStatus of 'Not Started' but they were NULL. Should we also se these to 'Not Started' or is this correct?

File: src/migrations/20231017230001-fei-goal-status-changes.js (hunk 1 above)
Comment (hardwarehuman) [reply src/migrations/20231017230001-fei-goal-status-changes.js:36]: There really shouldn't be any Goals on approved ARs that aren't through Objectives, but yes, we could go either way.

**Comment by hardwarehuman** [comment]
> In my DB (from a few weeks ago) I didn't see these goals linked to any AR's with the calculatedStatus of 'Not Started' but they were NULL. Should we also se these to 'Not Started' or is this correct?

All of these goals should be set to either Not Started or In Progress; any of these Goals having a status of NULL is a bug. I'll check now.

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**